mvc/src/main/java/com/interface21/webmvc/servlet/view/JsonView.java

```diff
@@ -1,14 +1,47 @@
 package com.interface21.webmvc.servlet.view;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.interface21.web.http.MediaType;
 import com.interface21.webmvc.servlet.View;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
 import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class JsonView implements View {
 
+    private final static Logger log = LoggerFactory.getLogger(JsonView.class);
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
     @Override
-    public void render(final Map<String, ?> model, final HttpServletRequest request, HttpServletResponse response) throws Exception {
+    public void render(final Map<String, ?> model,
+                       final HttpServletRequest request,
+                       final HttpServletResponse response) throws Exception {
+
+        response.setContentType(MediaType.APPLICATION_JSON_UTF8_VALUE);
+        response.setCharacterEncoding(StandardCharsets.UTF_8.name());
+
+        Object dataToWrite;
+
+        if (model != null && model.size() == 1) {
+            Object singleValue = model.values().iterator().next();
+            dataToWrite = validateNull(singleValue);
+        } else {
+            dataToWrite = model;
+        }
```

Comment:
개인적으로는 변수를 초기화하지 않은 상태로 두는 걸 주의하려고 하는 편인데요, 
이 변수가 어떤 값으로 언제 사용되는지를 코드를 읽으면서 계속 추적해야해서 가독성이 낮아진다고 느꼈어요. 값이 들어가지 않은채로 사용되진 않을까?하는 불안감도 있었구요~~

그래서 이 부분을 method로 추출하고, early return을 사용하면 dataToWrite라는 변수를 초기화하지 않은 상태 + 가변으로 둘 필요도 없고, else도 사용하지 않아도 되면서 코드의 가독성도 더 올라갈 것 같아요 !! 어떠신가용

Reply:
> 언제 사용되는지 코드를 읽으면서 계속 추적

이 부분이 확실히 불편하게 느껴지는 것 같네요. 값이 들어가지 않으면 어떻게 되는지도 추적해야하는 부분이 불편할 수도 있겠군요. 피드백 적용해보겠습니다.

감사합니다!

Reply:
넵 근데 여전히 Object  dataTowrite가 외부에 선언만 되어있는 것 같아요!!
```suggestion
        final Object dataToWrite = extractData(model);
        objectMapper.writeValue(response.getWriter(), dataToWrite);
```
이런식으로 바꾸면 더 좋을 것 같아요